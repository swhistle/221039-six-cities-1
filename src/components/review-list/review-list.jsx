import React from "react";
import PropTypes from "prop-types";

import {ReviewItemComponent} from "../review-item/review-item.jsx";

export class ReviewListComponent extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {reviewList} = this.props;

    return <React.Fragment>
      <section className="property__reviews reviews">
        <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">{reviewList.length}</span></h2>
        <ul className="reviews__list">
          <li className="reviews__item">
            {
              reviewList.length > 0 ? reviewList.map((reviewItem) => <ReviewItemComponent key={reviewItem.id} review={reviewItem}/>) : ``
            }
          </li>
        </ul>
      </section>
    </React.Fragment>;
  }
}

ReviewListComponent.propTypes = {
  reviewList: PropTypes.array
};
