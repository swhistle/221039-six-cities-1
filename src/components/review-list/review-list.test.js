import React from 'react';
import renderer from 'react-test-renderer';

import {ReviewListComponent} from "./review-list.jsx";

const TEST_REVIEW_LIST = [
  {
    id: 1,
    rating: 4,
    user: {
      id: 2,
      name: `user`
    },
    comment: `very good`,
    dateTime: `2019-05-08T14:13:56.569Z`
  }
];

it(`ReviewListComponent correct renders`, () => {
  const reviewList = renderer
    .create(
        <ReviewListComponent review={TEST_REVIEW_LIST}/>
    ).toJSON();

  expect(reviewList).toMatchSnapshot();
});
