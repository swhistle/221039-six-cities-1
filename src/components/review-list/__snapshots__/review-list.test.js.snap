// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewListComponent correct renders 1`] = `null`;
